File: distribution/python/dqops/client/models/import_tables_queue_job_parameters.py
from typing import Any, Dict, List, Type, TypeVar, Union, cast

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

T = TypeVar("T", bound="ImportTablesQueueJobParameters")


@_attrs_define
class ImportTablesQueueJobParameters:
    """
    Attributes:
        connection_name (Union[Unset, str]): Connection name
        schema_name (Union[Unset, str]): Schema name
        table_name_contains (Union[Unset, str]): Optional filter for the table names to import. The table names that are
            imported must contain a substring matching this parameter. This filter is case sensitive.
        table_names (Union[Unset, List[str]]): Optional list of table names inside the schema. When the list of tables
            is empty, all tables are imported.
        tables_import_limit (Union[Unset, int]): Optional parameter to configure the limit of tables that are imported
            from a single schema. Leave this parameter blank to use the default limit (300 tables).
    """

    connection_name: Union[Unset, str] = UNSET
    schema_name: Union[Unset, str] = UNSET
    table_name_contains: Union[Unset, str] = UNSET
    table_names: Union[Unset, List[str]] = UNSET
    tables_import_limit: Union[Unset, int] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        connection_name = self.connection_name
        schema_name = self.schema_name
        table_name_contains = self.table_name_contains
        table_names: Union[Unset, List[str]] = UNSET
        if not isinstance(self.table_names, Unset):
            table_names = self.table_names

        tables_import_limit = self.tables_import_limit

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if connection_name is not UNSET:
            field_dict["connectionName"] = connection_name
        if schema_name is not UNSET:
            field_dict["schemaName"] = schema_name
        if table_name_contains is not UNSET:
            field_dict["tableNameContains"] = table_name_contains
        if table_names is not UNSET:
            field_dict["tableNames"] = table_names
        if tables_import_limit is not UNSET:
            field_dict["tablesImportLimit"] = tables_import_limit

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        d = src_dict.copy()
        connection_name = d.pop("connectionName", UNSET)

        schema_name = d.pop("schemaName", UNSET)

        table_name_contains = d.pop("tableNameContains", UNSET)

        table_names = cast(List[str], d.pop("tableNames", UNSET))

        tables_import_limit = d.pop("tablesImportLimit", UNSET)

        import_tables_queue_job_parameters = cls(
            connection_name=connection_name,
            schema_name=schema_name,
            table_name_contains=table_name_contains,
            table_names=table_names,
            tables_import_limit=tables_import_limit,
        )

        import_tables_queue_job_parameters.additional_properties = d
        return import_tables_queue_job_parameters

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties

File: dqops/src/main/java/com/dqops/core/jobqueue/jobs/table/ImportTablesQueueJobParameters.java
/*
 * Copyright © 2021-Present DQOps, Documati sp. z o.o. (support@dqops.com)
 *
 * This file is licensed under the Business Source License 1.1,
 * which can be found in the root directory of this repository.
 *
 * Change Date: This file will be licensed under the Apache License, Version 2.0,
 * four (4) years from its last modification date.
 */
package com.dqops.core.jobqueue.jobs.table;

import com.dqops.utils.docs.generators.SampleStringsRegistry;
import com.dqops.utils.docs.generators.SampleValueFactory;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import lombok.EqualsAndHashCode;

import java.util.List;

/**
 * Parameters for the {@link ImportTablesQueueJob} job that imports selected tables from the source database.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@EqualsAndHashCode(callSuper = false)
public class ImportTablesQueueJobParameters {
    @JsonPropertyDescription("Connection name")
    private String connectionName;

    @JsonPropertyDescription("Schema name")
    private String schemaName;

    @JsonPropertyDescription("Optional filter for the table names to import. The table names that are imported must contain a substring matching this parameter. This filter is case sensitive.")
    private String tableNameContains;

    @JsonPropertyDescription("Optional list of table names inside the schema. When the list of tables is empty, all tables are imported.")
    private List<String> tableNames;

    @JsonPropertyDescription("Optional parameter to configure the limit of tables that are imported from a single schema. Leave this parameter blank to use the default limit (300 tables).")
    private Integer tablesImportLimit;

    public ImportTablesQueueJobParameters() {
    }

    /**
     * Creates a schema import job parameters.
     * @param connectionName Connection name to import.
     * @param schemaName Schema name in the source database to import.
     * @param tableNameContains Optional filter for the table names to import.
     * @param tableNames Names of tables to import.
     */
    public ImportTablesQueueJobParameters(String connectionName, String schemaName, String tableNameContains, List<String> tableNames) {
        this.connectionName = connectionName;
        this.schemaName = schemaName;
        this.tableNameContains = tableNameContains;
        this.tableNames = tableNames;
    }

    /**
     * Returns the connection name to be imported.
     * @return Connection name.
     */
    public String getConnectionName() {
        return connectionName;
    }

    /**
     * Sets the connection name to be imported.
     * @param connectionName Connection name.
     */
    public void setConnectionName(String connectionName) {
        this.connectionName = connectionName;
    }

    /**
     * Returns the source schema name.
     * @return Source schema name.
     */
    public String getSchemaName() {
        return schemaName;
    }

    /**
     * Sets the source schema name.
     * @param schemaName Source schema name.
     */
    public void setSchemaName(String schemaName) {
        this.schemaName = schemaName;
    }

    /**
     * Returns an optional filter for the table names to return.
     * @return Optional filter for the table names.
     */
    public String getTableNameContains() {
        return tableNameContains;
    }

    /**
     * Sets an optional filter for the table names.
     * @param tableNameContains Table name filter.
     */
    public void setTableNameContains(String tableNameContains) {
        this.tableNameContains = tableNameContains;
    }

    /**
     * Returns the names of tables to import.
     * @return Table names to import.
     */
    public List<String> getTableNames() {
        return tableNames;
    }

    /**
     * Sets the names of tables to import.
     * @param tableNames Table names to import.
     */
    public void setTableNames(List<String> tableNames) {
        this.tableNames = tableNames;
    }

    /**
     * Returns the limit of tables imported from a schema.
     * @return Limit of tables or null if using the default configuration.
     */
    public Integer getTablesImportLimit() {
        return tablesImportLimit;
    }

    /**
     * Sets the limit of tables imported from a schema.
     * @param tablesImportLimit New limit of tables.
     */
    public void setTablesImportLimit(Integer tablesImportLimit) {
        this.tablesImportLimit = tablesImportLimit;
    }

    public static class ImportTablesQueueJobParametersSampleFactory implements SampleValueFactory<ImportTablesQueueJobParameters> {
        @Override
        public ImportTablesQueueJobParameters createSample() {
            return new ImportTablesQueueJobParameters() {{
                setConnectionName(SampleStringsRegistry.getConnectionName());
                setSchemaName(SampleStringsRegistry.getSchemaName());
                setTableNames(List.of(SampleStringsRegistry.getTableName()));
            }};
        }
    }
}
